testing/src/main/java/com/linecorp/decaton/testing/processor/ReprocessAwareOrdering.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.testing.processor;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class ReprocessAwareOrdering implements ProcessingGuarantee {
```

Comment:
Exactly.
> I think the intention of "Reprocess" here is that when fail-over occurs offset 1 could be re-processed by different instance which causes it to be processed after 2

I prefixed this guarantee with `ReprocessAware` because of following reason:
- If we support `at most once` semantics in the future, the ordering will be checked "strictly" (i.e. 1 never appears after 2)
- Then, we call such (strict) ordering as "ProcessOrdering" I think?

Reply:
I see?  I was thinking that we'll just add a test checking exactly no duplicate in delivery for at-most-once case but apparently your test can also be added. Fine with as-is then. Please just add and enhance javadoc for it with much detailed explanation of its semantic.

Reply:
Hmm but I agree with this point
https://github.com/line/decaton/pull/47#discussion_r436447109
> On the other hand it sounds bit wired to me that this framework has a built-in check for at-most-once because it's a "feature" outside of core functionality

Then it would be fine to name this as just `ProcessOrdering `.